version.py

```diff
@@ -0,0 +1,76 @@
+import argparse
+from fileinput import FileInput
+import re
+import sys
+from packaging import version
+import semver
+
+
+def get_pyright_version() -> str:
+    with open('pyright/__init__.py') as f:
+        match = re.search(
+            r'^__pyright_version__\s*=\s*[\'"]([^\'"]*)[\'"]', f.read(), re.MULTILINE
+        )
+        if not match:
+            raise RuntimeError('version is not set')
+
+        return match.group(1)
+
+
+def compare(ver: str) -> bool:
+    return version.parse(get_pyright_version()) < version.parse(ver)
+
+
+def set_pyright_ver(ver: str):
+    with FileInput('pyright/__init__.py', inplace=True) as f:  # type: ignore
+        for line in f:  # type: ignore
+            line = re.sub(
+                r'^__pyright_version__\s*=\s*[\'"]([^\'"]*)[\'"]',
+                f"__pyright_version__ = '{ver}'",
+                line.rstrip(),  # type: ignore
+            )  # type: ignore
+            print(line)
+
+
+def bump_pyright_package_ver():
+    with FileInput('pyright/__init__.py', inplace=True) as f:  # type: ignore
+        for line in f:  # type: ignore
+            line = line.rstrip()  # type: ignore
+            current_version = re.search(
+                r'^__version__\s*=\s*[\'"]([^\'"]*)[\'"]', line  # type:ignore
+            )
+            if current_version:
+                line = f"__version__ = '{semver.VersionInfo.parse(current_version.group(1)).bump_patch()}'"  # type:ignore
+            print(line)  # type: ignore
```

Comment:
These `# type: ignore` comments can be removed by switching from `FileInput` to `fileinput.input()`
```diff
- with FileInput('pyright/__init__.py', inplace=True) as f:
+ with fileinput.input('pyright/__init__.py', inplace=True) as f
```

Reply:
Ah, i knew there had to be a simple way.